docs/proposals/PROPOSAL_A.md

```diff
@@ -0,0 +1,203 @@
+
+| Jail parameter | JSON equivalent      |
+| -------------- | -------------------- |
+| jid            | -                    |
+| name           | see below            |
+| path           | root.path            |
+| ip4.addr       | freebsd.jail.ip4Addr |
+| ip4.saddrsel   | -                    |
+| ip4            | freebsd.jail.ip4     |
+| ip6.addr       | freebsd.jail.ip6Addr |
+| ip6.saddrsel   | -                    |
+| ip6            | freebsd.jail.ip6     |
+| vnet           | freebsd.jail.vent    |
+| host.hostname  | hostname             |
+| host           | freebsd.jail.host    |
+| sysvmsg        | freebsd.jail.sysvmsg |
+| sysvsem        | freebsd.jail.sysvsem |
+| sysvshm        | freebsd.jail.sysvshm |
+| securelevel    | -                    |
+| devfs_ruleset  | see below            |
+| children.max   | see below            |
+| enforce_statfs | freebsd.jail.enforceStatfs |
+| persist        | -                    |
+| parent         | freebsd.jail.parent  |
+| osrelease      | -                    |
+| osreldate      | -                    |
+| allow.set_hostname | freebsd.jail.allow.setHostname |
+| allow.sysvipc  | freebsd.jail.allow.sysvipc |
+| allow.raw_sockers  | freebsd.jail.allow.rawSockets |
+| allow.chflags  | freebsd.jail.allow.chflags |
+| allow.mount    | freebsd.jail.allow.mount |
+| allow.quotas    | freebsd.jail.allow.quotas |
+| allow.read_msgbuf | freebsd.jail.allow.readMsgbuf |
+| allow.socket_af | freebsd.jail.allow.socketAf |
+| allow.mlock    | freebsd.jail.allow.mlock |
+| allow.nfsd     | freebsd.jail.allow.nfsd |
+| allow.reserved_ports | freebsd.jail.allow.reservedPorts |
+| allow.unprivileged_proc_debug | - |
+| allow.suser    | freebsd.jail.allow.suser |
+| allow.mount.*  | see below            |
+| securelevel    | -                    |
+
+The jail name is set to the create command's `container-id` argument.
+
+The `devfs_ruleset` parameter is only required for jails which create new `devfs` mounts - typically OCI runtimes will mount `devfs` on the host.
+
+The `children.max` parameter is managed by the OCI runtime e.g when a new container shares namespaces with an existing container.
+
+The `allow.mount.*` parameter set is extensible - this proposal suggests representing allowed mount types as an array. As with `devfs`, typically the OCI runtime will manage mounts for the container by performing mount operations on the host.
+
+### Example
+
+An example config for a container with its own host and network namespaces. This
+container is allowed to see its own mounts and can use raw
+sockets. In addition to the minimal set of devices in the container devfs,
+`/dev/pf` is exposed, allowing the container to manage firewall rules etc. in its
+network namespace.
+
+```json
+{
+	"ociVersion": "1.1.0",
+	"hostname": "mycontainer",
+	"process": {
+		"cwd": "/",
+		"env": ["PATH=/bin:/sbin:/usr/bin:/usr/sbin"],
+		"args": ["freebsd-version"]
```

Comment:
This should probably be a fully-qualified path; with this bundle there wouldn't be a shell to do PATH resolution.

Reply:
The shell doesn't do PATH resolution - this usually happens in the execvp library function. Linux runtimes that I've looked at perform PATH resolution in the runtime to improve error reporting for typos or missing binaries and this is the direction I took in ocijail.